blogging/views.py

```diff
@@ -22,13 +24,23 @@ def list_view(request):
     # body = template.render(context)
     context = {'posts': posts}
     return render(request, 'blogging/list.html', context)
+
+class BlogListView(ListView):
+    model = Post
+    template_name = 'blogging/list.html'
+    queryset = Post.objects.order_by('-published_date').exclude(published_date__exact=None)
+
+class BlogDetailView(DetailView):
```

Comment:
This one is actually simpler:

```
class BlogDetailView(DetailView):
    queryset = Post.objects.exclude(published_date__exact=None)
    template_name = "blogging/detail.html"
```

The above will provide the template with a _post_ object to work with.